src/Resolver/Directory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Upward\Resolver;
+
+use Magento\Upward\Definition;
+use Zend\Http\Header\ContentType;
+use Zend\Http\Response;
+use Zend\Http\Response\Stream;
+
+class Directory extends AbstractResolver
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getIndicator(): string
+    {
+        return 'directory';
+    }
+
+    public function isValid(Definition $definition): bool
+    {
+        $directory  = $this->getIterator()->get('directory', $definition);
+        $upwardRoot = $definition->getBasepath();
+
+        $root = realpath($upwardRoot . \DIRECTORY_SEPARATOR . $directory);
+
+        if (!$root || !is_dir($root)) {
+            return false;
+        }
+
+        return parent::isValid($definition);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($definition)
+    {
+        $directory = $definition;
+
+        if ($definition instanceof Definition) {
```

Comment:
I've seen this check before, and was just curious what the use case is. Even with the shorthand config below, it still comes through as a Definition.

```
body:
  directory: './templates'
```

I'm not a fan of immediate re-assignment, but I would say optionally I would add a ternary here if this needs to be left as-is.

Reply:
Honestly, this is probably me over-engineering things and not feeling totally comfortable that we can't just typehint `Definition` here.

In concept I thought it could theoretically be possible for a string (the value of `directory`) to be passed to `resolve()` and us be able to handle it appropriately

In practice though, there isn't any way this could be done so it doesn't really make any sense.

Reply:
Fair enough, have to deal with that mixed type until we can revisit the scalar/Definition logic in DefinitionIterator. Some re-factoring might need to be done in some of my resolvers, I think all of mine assume it's receiving a Definition.